papaparse.js

```diff
@@ -1335,7 +1335,7 @@ License: MIT
 					avgFieldCount /= (preview.data.length - emptyLinesCount);
 
 				if ((typeof bestDelta === 'undefined' || delta <= bestDelta)
-					&& (typeof maxFieldCount === 'undefined' || avgFieldCount > maxFieldCount) && avgFieldCount > 1.99) {
+					&& ((typeof maxFieldCount === 'undefined' || avgFieldCount > maxFieldCount) || delta < bestDelta) && avgFieldCount > 1.99) {
```

Comment:
This is already tested in the line avobe. So for me this condition is redundant.

Reply:
The problem is, that if delta is smaller or equal bestDelta, avgFieldCount needs to be bigger than maxFieldCount which might be not the case if e.g. comma appears more often but delimiter is another sign. Therefore I added the case that if avgFieldCount isn't bigger but delta is smaller, the delimiter is more accurate.

Reply:
This sound so much complicated and I prefer to keep it simple (or at least what it is now)

I do not like so much the current solution as we will always end up with a case where is not supported which requires PapaParse to be modified. 

So I'm wondering if it won't be better to allow the end user to pass a function that accepts the current parameters and returns true or false to determine if the delimiter is accepted or not.

PapaParse should have a default function (which should be callable from user custom functions) that will be used if no user defined function.

What do you think?

Reply:
I'm not sure if we talk about the same topic. 
What do you mean with "current solution": mine or the one in repo right now?

I'm not talking about accepted delimiters or not, I just had a file where it guessed the wrong delimiter because in it (see #804 for example file) there were more commas included (due to geometry objects which are separated by comma) than semicolons (which in this case is delimiter as you can see in header).
The reason for this was that avgFieldCount for comma was of course higher because it ended in more fileds. But the delta was 4 because the header was considered as one field.

So when guessing the right delimiter I think we could approach that a smaller delta is always better if avgFieldCount is bigger than 2. 
The comparison avgFieldCount > maxFieldCount for me makes only sense if deltas are equal.
So maybe the comparison suggested in my comment above is less redundant and does the same job.

But maybe I made an error in reasoning, I'm not sure. I even didn't test the approach in comment above yet.

Reply:
"current solution" means the current one and it is exactly the same design as your proposal.
What I talk about is that we had so much request to adapt the delimiter guessing functionality to specific case and that's why I think we should improve the current design instead of adding support another special case.

Reply:
Okay, got it. If there were already many requests adapting the guessing, I understand that you want to rethink current design.
My intention was to fix the current approach, because I thought its not that bad except this problem. 
But sure with different files, there might be different problems  where even this implementation might fail.

So your idea is to provide end user a function somehow like guessDelimiter() which returns true or false?

Reply:
The guess delimiter function is currenctly configurable but it sems that there is not so much people using this feature. 
So in order to improve it's adoption I think we need a way to call the standard function from the custom function.

Reply:
By configurable do you mean the arguments you pass to guess delimiter function which are based on config or is there another way to configure it?
Should the default function which PapaParse uses, if there is no custom function, be like the current guess delimiter function?